pkg/ingester/client/cache.go

```diff
@@ -0,0 +1,89 @@
+package client
+
+import (
+	io "io"
+	"sync"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/weaveworks/cortex/pkg/util"
+)
+
+// Factory defines the signature for an ingester client factory
+type Factory func(addr string, cfg Config) (IngesterClient, error)
+
+// IngesterClientCache holds a cache of ingester clients
+type IngesterClientCache struct {
```

Comment:
Is "cache" the right word?  Elsewhere I've seen this called a "connection pool", except we have max one connection per endpoint.  Maybe just some more explanation of the intended uses would help.

Reply:
I like IngesterPool, I will change it to that, and add some more explanations.